pkg/test/resourcefixture/contexts/register.go

```diff
@@ -452,16 +452,19 @@ func directDelete(ctx context.Context, u *unstructured.Unstructured, c client.Cl
 	// in subsequent calls but found it originally, we can consider it deleted.
 	firstFind := false
 
+	var deleteError error
 	for i := 0; i < maxRetries; i++ {
 		found, err := a.Find(ctx)
-		if err == nil && found {
+		deleteError = err
+
+		if deleteError == nil && found {
 			firstFind = true
 			op := directbase.NewDeleteOperation(c, u)
-			_, err = a.Delete(ctx, op)
-			if err == nil {
+			_, deleteError = a.Delete(ctx, op)
+			if deleteError == nil {
 				return nil // success
 			}
```

Comment:
Suggest something like this for readability (the same pattern as the above call): 

```
_, err := a.Delete(ctx, op)
lastErr = err
if err == nil {
```